src/montgomery.rs

```diff
@@ -150,34 +193,256 @@ impl CompressedMontgomeryU {
     }
 }
 
+/// A point on the Montgomery form of the curve, in projective 𝗣^2 coordinates.
+///
+/// The transition between affine and projective is given by
+///
+///      u → U/W
+///      v → V/W
+///
+/// thus the Montgomery curve equation
+///
+///      E_(A,B) : Bv² = u(u² + Au + 1)
+///
+/// becomes
+///
+///      E_(A,B) : BV²W = U(U² + AUW + W²) ⊆ 𝗣^2
+///
+/// Here, again, to differentiate from points in the twisted Edwards model, we
+/// call the point `(x,y)` in affine coordinates `(u,v)` and similarly in projective
+/// space we use `(U:V:W)`.  However, since (as per Montgomery's original work) the
+/// v-coordinate is superfluous for the purposes of scalar multiplication, we merely
+/// use `(U:W)`.
+#[derive(Copy, Clone, Debug)]
+#[allow(missing_docs)]
+pub struct MontgomeryPoint{
+    pub U: FieldElement,
+    pub W: FieldElement,
+}
+
+/// The identity point is a unique point (the only where `W = 0`) on the curve.
+///
+/// In projective coordinates, the quotient map `x : E (A,B) → E/<⦵> = 𝗣¹` is
+///
+///               ⎧ (x_P:1) if P = (x_P:y_P:1) ,
+///      x : P ↦  ⎨
+///               ⎩   (1:0) if P = O = (0:1:0) .
+///
+/// We emphasize that the formula `x((U: V : W)) = (U : W)` only holds on the
+/// open subset of `E_(A,B)` where `W ≠ 0`; it does not extend to the point
+/// `O = (0:1:0)` at infinity, because `(0:0)` is not a projective point.
+///
+/// # Returns
+///
+/// The (exceptional) point at infinity in the Montgomery model.
+impl Identity for MontgomeryPoint {
+    fn identity() -> MontgomeryPoint {
+        MontgomeryPoint {
+            U: FieldElement::one(),
+            W: FieldElement::zero(),
+        }
+    }
+}
+
+/// Determine if two `MontgomeryPoint`s are equal, in constant time.
+///
+/// # Note
+///
+/// Because a compressed point on the Montgomery form of the curve doesn't
+/// include the sign bit, there's two points here (if translated from the
+/// Edwards form) which will equate.
+///
+/// # Returns
+///
+/// `1` if the points are equal, and `0` otherwise.
+impl Equal for MontgomeryPoint {
+    fn ct_eq(&self, that: &MontgomeryPoint) -> u8 {
+        slices_equal(self.compress().as_bytes(),
+                     that.compress().as_bytes())
+    }
+}
+
+/// Determine if this `MontgomeryPoint` is valid.
+///
+/// # Note
+///
+/// All points, except for `(X:W) = (0:0)`, are valid, since the projective
```

Comment:
nit: I think this should just say that all projective points are valid, and `(0:0)` is the only point of `F_p^2` which does not represent a projective point.

Reply:
Okay I reworded it a bit in 7b378ada6